src/components/Button.tsx

```diff
@@ -7,11 +7,16 @@ type NativeButtonType = ComponentProps<"button">;
 
 type ButtonProps = {
   type?: NativeButtonType["type"];
+  onClick?: NativeButtonType["onClick"];
   children?: ReactNode;
 };
```

Comment:
Button without onClick makes no sense, make it required. What you probably want is to join the two types together

```suggestion
type ButtonProps = {
  type?: NativeButtonType["type"];
  onClick?: NativeButtonType["onClick"];
  children?: ReactNode;
} & Omit< NativeButtonType, 'onClick'>
```

and Omit the props that you are overriding. I believe button has optional onClick, since we want to make it required, we need to first, remove it from the native (if we will join them like suggested above) and then override it with our version

Reply:
resolved